velox/functions/remote/client/Remote.cpp

```diff
@@ -71,7 +90,41 @@ class RemoteFunction : public exec::VectorFunction {
   }
 
  private:
-  void applyRemote(
+  void applyRestRemote(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const {
+    try {
+      serializer::presto::PrestoVectorSerde serde;
+      auto remoteRowVector = std::make_shared<RowVector>(
+          context.pool(),
+          remoteInputType_,
+          BufferPtr{},
+          rows.end(),
+          std::move(args));
+
+      std::unique_ptr<folly::IOBuf> requestBody =
+          std::make_unique<folly::IOBuf>(rowVectorToIOBuf(
+              remoteRowVector, rows.end(), *context.pool(), &serde));
+
+      std::unique_ptr<folly::IOBuf> responseBody = restClient_->invokeFunction(
+          boost::get<std::string>(metadata_.location), std::move(requestBody));
+
+      auto outputRowVector = IOBufToRowVector(
+          *responseBody, ROW({outputType}), *context.pool(), &serde);
+
+      result = outputRowVector->childAt(0);
+    } catch (const std::exception& e) {
```

Comment:
Is it possible that some of the rows encountered errors while some not ? Seems like Thrift is handling that case specially to set the errors context.